packages/integration-aws-node/api-report.md

```diff
@@ -0,0 +1,73 @@
+## API Report File for "@backstage/integration-aws-node"
+
+> Do not edit this file. It is a report generated by [API Extractor](https://api-extractor.com/).
+
+```ts
+import { AwsCredentialIdentityProvider } from '@aws-sdk/types';
+import { Config } from '@backstage/config';
+
+// @public
+export type AwsCredentials = {
+  accountId?: string;
+  stsRegion?: string;
+  provider: AwsCredentialIdentityProvider;
+};
+
+// @public
+export interface AwsCredentialsProvider {
+  getCredentials(opts?: AwsCredentialsProviderOptions): Promise<AwsCredentials>;
+}
```

Comment:
Looking at the TechDocs code I realized that the naming here is potentially a bit confusing, as this is more like a `AwsCredentialsMetaProvider`. Especially the `getCredentials` method doesn't actually return credentials, but another provider. In isolation I don't think it's too much of an issue, but when compared to the `GithubCredentialsProvider` they don't line up very well. The GitHub one requires you to request credentials for every call that you make, i.e. it's the `AwsCredentialIdentityProvider` from `@aws-sdk/types`.

What do you think about some slight differentiation in naming between the two, so what the usage patterns are more clear? I'm assuming that the `AwsCredentialIdentityProvider` itself is used directly with the AWS SDK and we wouldn't want to move the implementation down to that level.

One idea that comes to mind is to rename the method to `getCredentialsProvider` or just `getProvider`, it's a bit strange when you consider the name of the interface but tbh I think that's going to look alright in code. That way we'd have two categories of credential providers, one that provide credentials directly, and another that operates one level up and typically provides you with a credentials object for the accompanying SDK. What do you think?

Reply:
Yeah I really dislike the naming as it's implemented :smile: As you pointed out, the class is really a provider of credential providers.  Ideally I'd like to come up with a more accurate class name.  Do any of these strike you as better?
* `AwsCredentialsIntegration.getCredentialsProvider`
* `AwsCredentialsManager.getCredentialsProvider`
* `AwsCredentialProviderManager.getCredentialsProvider`
* `AwsCredentialProviderResolver.getCredentialsProvider`

Right now I'm leaning towards AwsCredentialsManager

Reply:
Yeah same, `AwsCredentialsManager.getCredentialsProvider`. I'm feeling that'll work with any kind of credentials integration and it could be that we switch out the GitHub one to be `GithubCredentialsManager.getCredentials` for consistency. `AwsCredentialsIntegration` I feel has a risk of confusion that it's an integration towards an "AWS credentials" service, and the other ones I don't feel add much. CC the rest of @backstage/maintainers for 👀 too

Reply:
Renamed AwsCredentialsProvider.getCredentials -> AwsCredentialsManager.getCredentialProvider